transport/src/main/java/io/netty/channel/nio/NioEventLoop.java

```diff
@@ -486,10 +551,12 @@ private void processSelectedKeysPlain(Set<SelectionKey> selectedKeys) {
                 }
             }
         }
+        return processed;
     }
 
-    private void processSelectedKeysOptimized(SelectionKey[] selectedKeys) {
-        for (int i = 0;; i ++) {
+    private int processSelectedKeysOptimized(SelectionKey[] selectedKeys) {
+        int i = 0;
+        for (;; i ++) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) This loop's stop condition tests "" but the incrementer updates "i". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1994)